tools/tests/test_html_branches.py

```diff
@@ -112,15 +113,25 @@ def test_build_id_dict(self):
 
         template_id_dict = build_id_dict(templates)
 
-        self.assertEqual(list(template_id_dict.keys()), ['hello_{{ message }}', 'intro', 'below_navbar'])
-        self.assertEqual(template_id_dict['hello_{{ message }}'], [
-                         'Line 12:/srv/zulip/tools/tests/test_template_data/test_template1.html',
-                         'Line 12:/srv/zulip/tools/tests/test_template_data/test_template2.html'])
-        self.assertEqual(template_id_dict['intro'], [
-                         'Line 10:/srv/zulip/tools/tests/test_template_data/test_template1.html',
-                         'Line 11:/srv/zulip/tools/tests/test_template_data/test_template1.html',
-                         'Line 11:/srv/zulip/tools/tests/test_template_data/test_template2.html'])
-        self.assertEqual(template_id_dict['below_navbar'], ['Line 10:/srv/zulip/tools/tests/test_template_data/test_template2.html'])
+        self.assertEqual(sorted(list(template_id_dict.keys())), ['below_navbar', 'hello_{{ message }}', 'intro'])
+
+        self.assertEqual(len(template_id_dict['hello_{{ message }}']), 2)
+        self.assertTrue(re.match(re.escape('Line 12:') + '(.+)' + re.escape('/tools/tests/test_template_data/test_template1.html'),
+                                 template_id_dict['hello_{{ message }}'][0]))
+        self.assertTrue(re.match(re.escape('Line 12:') + '(.+)' + re.escape('/tools/tests/test_template_data/test_template2.html'),
+                                 template_id_dict['hello_{{ message }}'][1]))
+
+        self.assertEqual(len(template_id_dict['intro']), 3)
+        self.assertTrue(re.match(re.escape('Line 10:') + '(.+)' + re.escape('/tools/tests/test_template_data/test_template1.html'),
+                                 template_id_dict['intro'][0]))
+        self.assertTrue(re.match(re.escape('Line 11:') + '(.+)' + re.escape('/tools/tests/test_template_data/test_template1.html'),
+                                 template_id_dict['intro'][1]))
+        self.assertTrue(re.match(re.escape('Line 11:') + '(.+)' + re.escape('/tools/tests/test_template_data/test_template2.html'),
+                                 template_id_dict['intro'][2]))
+
+        self.assertEqual(len(template_id_dict['below_navbar']), 1)
+        self.assertTrue(re.match(re.escape('Line 10:') + '(.+)' + re.escape('/tools/tests/test_template_data/test_template2.html'),
+                                 template_id_dict['below_navbar'][0]))
```

Comment:
I wonder if it'd be cleaner to do `x.startswith("Line 10") and x.endswith("tools/tests/test_template_data/test_template1.html") rather than using the `re` module.